### PR TITLE
FAQ: remove errors

### PR DESCRIPTION
**Pull request by maryamsulemani97**
part of #1593 

This PR moves the error types FAQ to API reference/overview

### CODE REVIEW

File: reference/api/overview.md [hunk 1]
```diff
@@ -38,27 +38,25 @@ Please read about [security keys](/learn/security/master_api_keys.md) and [how t
...
+| 201         | ✅ **Created** The resource has been created (synchronous)                                 |
+| 202         | ✅  **Accepted** The task has been added to the queue (asynchronous)                       |
+| 204         | ✅ **No Content** The resource has been deleted or no content has been returned            |
+| 205         | ✅ **Reset Content** All the resources have been deleted                                   |
+| 400         | ❌ **Bad Request** The request was unacceptable, often due to missing a required parameter |
+| 401         | ❌ **Unauthorized** No valid API key provided                                              |
+| 403         | ❌ **Forbidden** The API key doesn't have the permissions to perform the request           |
+| 404         | ❌ **Not Found** The requested resource doesn't exist                                      |
+
+### Error types
```
Comment (guimachiavelli) [inline reference/api/overview.md:53]: I think the "Error types" section requires a bit more context. These errors are related but separate from 4xx HTTP status codes. What do you think about something like this:

```suggestion
### Errors

Failed requests are always accompanied by a JSON-formatted error response. Meilisearch errors can be of one of the following types:
```

File: reference/api/overview.md [hunk 2]
```diff
@@ -38,27 +38,25 @@ Please read about [security keys](/learn/security/master_api_keys.md) and [how t
...
+
+| Type              | Description                                                                                                 |
+|-------------------|-------------------------------------------------------------------------------------------------------------|
+| `invalid_request` | This is due to an error in the user input. It is accompanied by the HTTP code `4xx`                         |
+| `internal`        | This is due to machine or configuration constraints. The most common cause is reaching or exceeding hard limits, such as the size of the disk, the size limit of an index, etc. It is accompanied by the HTTP code `5xx`  |
+| `auth`            | This type of error is related to authentication and authorization. It is accompanied by the HTTP code `4xx` |
 
 All errors contain a JSON body that explains the error.
```
Comment (guimachiavelli) [inline reference/api/overview.md:61]: Moved this info above the table.

```suggestion
```

**Review by guimachiavelli — COMMENTED** [review]
I think the changes are good, but felt the content structure became a bit unclear with the new content.

File: reference/api/overview.md [hunk 3]
```diff
@@ -38,27 +38,25 @@ Please read about [security keys](/learn/security/master_api_keys.md) and [how t
 
 ## Errors & status code
```
[the remark this answers is not in the document]
Comment (guimachiavelli) [reply reference/api/overview.md:39]: Or this?

```suggestion
## API request status codes and Meilisearch errors
```

[the remark this answers is not in the document]
Comment (maryamsulemani97) [reply reference/api/overview.md:39]: I like first one better

**Review by guimachiavelli — APPROVED** [review]
:neckbeard:

**Comment by maryamsulemani97** [comment]
bors merge

**Pull request merged**